pkg/agent/profile.go

```diff
@@ -145,31 +146,35 @@ func (p *CgroupProfiler) LastError() error {
 }
 
 func (p *CgroupProfiler) Stop() {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
 	level.Debug(p.logger).Log("msg", "stopping cgroup profiler")
 	if p.cancel != nil {
 		p.cancel()
 	}
 }
 
-func (p *CgroupProfiler) Labels() []*profilestorepb.Label {
-	labels := append(p.target.Labels(),
-		&profilestorepb.Label{
-			Name:  "__name__",
-			Value: "parca_agent_cpu",
-		})
-	for key, value := range p.externalLabels {
-		labels = append(labels, &profilestorepb.Label{
-			Name:  key,
-			Value: value,
-		})
+func (p *CgroupProfiler) Labels() model.LabelSet {
+	labels :=
+		model.LabelSet{
+			"__name__": "parca_agent_cpu",
```

Comment:
I can see this hasn't been a constant before, but in a future Pull Request we can make this a constant too. :)
For the label there should also be `labels.MetricName` be available.

Reply:
It's just coincidental that we chose `__name__` as well, using `labels.MetricName` would be confusing I think.